rfc/rfc-79/rfc-79-2.md

```diff
@@ -0,0 +1,99 @@
+
+## Design
+### Enabling a plan to be pre-emptable
+To satisfy goal (1), a new config flag "cancellable" can be added to a table service plan. A writer that intends to schedule a cancellable table service plan can enable the flag in the serialized plan metadata. Any writer executing the plan can infer that the plan is cancellable, and when trying to commit the instant should abort if it detects that any ingestion write or table service plan (without cancellable config flag) is targeting the same file groups. On the other side, the commit finalization flow for ingestion writers can be updated to ignore any inflight table service plans if they are cancellable.
+For the purpose of this design proposal, consider an ingestion job as having three steps:
+1. Schedule itself on the timeline with a new instant time in a .requested file
+2. Process/record tag incoming records, build a workload profile, and write the updating/replaced file groups to a "inflight" instant file on the timeline. Check for conflicts and abort if needed.
+3. Perform write conflict checks and commit the instant on the timeline
+
+The aforementioned changes to ingestion and table service flow will ensure that in the event of a conflicting ingestion and cancellable table service writer, the ingestion job will take precedence unless the table service job was completed before (2). Since in this scenario the ingestion job will see that a completed instant (a cancellable table service action) conflicts with its ongoing inflight write, and therefore it would not be legal to proceed. Unfourtatnely this means that this design cannot compeletly guarantee that ingestion job will always take precedence. But future enhancements/hueristics can be explored to descrease the chance of this scenario, such as
+* Have the ingestion writer write a "hint" of possible partitions it might affect in the .requested file, and the cancellable table service writer can check that before commiting the table service plan
```

Comment:
We already have early conflict detection support. we can re-use some of the logic from there and not re-invent the wheel